backpack/utils/conv_transpose.py

```diff
@@ -40,15 +32,7 @@ def extract_weight_diagonal(module, input, grad_output, N):
 
     input_reshaped = input.reshape(M, -1, spatial_out_numel)
 
-    if N == 1:
-        grad_output_viewed = grad_output
-    elif N == 2:
-        grad_output_viewed = eingroup("v,n,c,h,w->v,n,c,hw", grad_output)
-    elif N == 3:
-        grad_output_viewed = eingroup("v,n,c,d,h,w->v,n,c,dhw", grad_output)
-    else:
-        raise ValueError("{}-dimensional ConvTranspose is not implemented.".format(N))
-
+    grad_output_viewed = rearrange(grad_output, "v n c ... -> v n c (...)")
```

Comment:
This looks exactly identical to what `separate_channels_and_pixels` in `utils/conv.py` does, so we could recycle the function.